File: backend/app/schemas/minuta.py
from pydantic import BaseModel


class MinutaDecision(BaseModel):
    pregunta: str
    opcion_a: str
    opcion_b: str
    decision_tomada: str | None = None


class MinutaCompromiso(BaseModel):
    descripcion: str
    fecha: str


class MinutaTema(BaseModel):
    id: int
    titulo: str
    sintesis: str
    decision: MinutaDecision
    compromiso: MinutaCompromiso | None = None


class MinutaOut(BaseModel):
    generada: bool
    carta: str
    temas: list[MinutaTema]


class DecisionIn(BaseModel):
    tema_id: int
    decision: str

File: frontend/src/lib/minuta.ts
import api from "@/lib/api"

export interface MinutaDecision {
  pregunta: string
  opcion_a: string
  opcion_b: string
  decision_tomada: string | null
}

export interface MinutaCompromiso {
  descripcion: string
  fecha: string
}

export interface MinutaTema {
  id: number
  titulo: string
  sintesis: string
  decision: MinutaDecision
  compromiso: MinutaCompromiso | null
}

export interface MinutaOut {
  generada: boolean
  carta: string
  temas: MinutaTema[]
}

export async function getMinuta(): Promise<MinutaOut> {
  const r = await api.get<MinutaOut>("/annual-plan/minuta")
  return r.data
}

export async function sesionarConsejo(): Promise<MinutaOut> {
  const r = await api.post<MinutaOut>("/annual-plan/minuta")
  return r.data
}

export async function cerrarDecision(temaId: number, decision: "A" | "B" | "aplazar"): Promise<MinutaOut> {
  const r = await api.post<MinutaOut>("/annual-plan/minuta/decision", { tema_id: temaId, decision })
  return r.data
}
